### PR TITLE
add skeleton of bootnode

### PR DESCRIPTION
**Pull request by LeoHChen**
support save/load private key to .bnkey file

Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

**Comment by LeoHChen** [comment]
```
[20:00] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ ./scripts/go_executable_build.sh bootnode
building cmd/bootnode/main.go
Harmony (C) 2019. bootnode, version v2029-e5beed1 (ec2-user@ 2019-01-28T20:01:16+0000)
Exit status: 1 [20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
INFO [01-28|20:01:21.394] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmdestnbJj6SNYHsY1fYk7Tbt1jZ1nEYDN8DRPz7vtCRkz
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ rm .bnkey
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
WARN [01-28|20:01:30.367] No priviate key can be loaded from file  keyfile=./.bnkey
WARN [01-28|20:01:30.367] Using random private key
INFO [01-28|20:01:30.777] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
INFO [01-28|20:01:32.970] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf
```

**Comment by LeoHChen** [comment]
```
[20:00] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ ./scripts/go_executable_build.sh bootnode
building cmd/bootnode/main.go
Harmony (C) 2019. bootnode, version v2029-e5beed1 (ec2-user@ 2019-01-28T20:01:16+0000)
Exit status: 1 [20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
INFO [01-28|20:01:21.394] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmdestnbJj6SNYHsY1fYk7Tbt1jZ1nEYDN8DRPz7vtCRkz
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ rm .bnkey
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
WARN [01-28|20:01:30.367] No priviate key can be loaded from file  keyfile=./.bnkey
WARN [01-28|20:01:30.367] Using random private key
INFO [01-28|20:01:30.777] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf
[20:01] ec2-user @ leo-cloud: ~/go/src/github.com/harmony-one/harmony [bootnode]: Up-to-date
$ bin/bootnode
INFO [01-28|20:01:32.970] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf
```

File: cmd/bootnode/main.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+		panic(err)
+	}
+
+	opts = append(opts, libp2p.Identity(privKey))
+
+	host, err := libp2p.New(context.Background(), opts...)
+	if err != nil {
+		panic(err)
+	}
+
```
Comment (alajko) [inline cmd/bootnode/main.go]: Does the host need to be "started"?

Comment (LeoHChen) [reply cmd/bootnode/main.go]: not yet.  this is just the first step.

**Comment by LeoHChen** [comment]
TEST

```
$ bin/bootnode
DEBUG[01-29|00:16:10.118] HostV2 is up!                            port=9876 id=QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf addr=/ip4/0.0.0.0/tcp/9876
INFO [01-29|00:16:10.118] NewHost                                  port=9876 ip=127.0.0.1 self=127.0.0.1:9876 PeerID="<peer.ID Qm*JgbVuf>"
INFO [01-29|00:16:10.118] bootnode                                 BN_MA=/ipv/127.0.0.1/tcp/9876/p2p/QmayB8NwxmfGE4Usb4H61M8uwbfc7LRbmXb3ChseJgbVuf
```

**Pull request merged**